srml/contracts/src/rent.rs

```diff
@@ -111,68 +123,54 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 	)
 	.is_ok();
 
-	if !rent_allowance_exceeded && can_withdraw_rent && !go_below_subsistence {
-		// Collect dues
-
-		if pay_rent {
-			let imbalance = T::Currency::withdraw(
-				account,
-				dues,
-				WithdrawReason::Fee,
-				ExistenceRequirement::KeepAlive,
-			)
-			.expect(
-				"Withdraw has been checked above;
-				go_below_subsistence is false and subsistence > existencial_deposit;
-				qed",
-			);
-
-			<ContractInfoOf<T>>::mutate(account, |contract| {
-				let contract = contract
-					.as_mut()
-					.and_then(|c| c.as_alive_mut())
-					.expect("Dead or inexistent account has been exempt above; qed");
-
-				contract.rent_allowance -= imbalance.peek(); // rent_allowance is not exceeded
-				contract.deduct_block = current_block_number;
-			})
-		}
+	if can_withdraw_rent && (go_below_subsistence || pay_rent) {
+		// Collect dues.
+		let _ = T::Currency::withdraw(
+			account,
+			dues_limited,
+			WithdrawReason::Fee,
+			ExistenceRequirement::KeepAlive,
+		)
+		.expect(
+			"Withdraw has been checked above;
+			dues_limited < rent_budget < balance - subsistence < balance - existential_deposit;
+			qed",
+		);
+	}
 
-		RentOutcome::Ok
-	} else {
-		// Evict
-
-		if can_withdraw_rent && !go_below_subsistence {
-			T::Currency::withdraw(
-				account,
-				dues,
-				WithdrawReason::Fee,
-				ExistenceRequirement::KeepAlive,
-			)
-			.expect("Can withdraw and don't go below subsistence");
-		} else if !is_below_subsistence {
-			T::Currency::make_free_balance_be(account, subsistence_threshold);
-		} else {
-			T::Currency::make_free_balance_be(account, <BalanceOf<T>>::zero());
-		}
+	if go_below_subsistence || !can_withdraw_rent {
```

Comment:
oh ok I see you replaced make_free_balance_be by just not withdrawing anything if it can't withdraw.

So now someone that can't withdraw its partial dues on tombstone creation will now not pay its dues on tombstone creation I feel ok with that. 

(as note when implementing this I think I got inspired by slashing function which I think bypass locks. But at the same time tombstone is more legit that slash so I prefer your way now)

Reply:
Yeah, I'm not sure this is the right behavior. I see three options:

1. Respect the locks, which is what the current PR does.
2. Allow `withdraw` to bypass locks when given `WithdrawalReason::Fee`.
3. Couple `LockableCurrency` with `rent_allowance`. But despite seemingly giving the right behavior, I think it would get really complicated really fast.